examples/elasticsearch_helper_example/src/Plugin/ElasticsearchIndex/MultilingualContentIndex.php

```diff
@@ -102,29 +103,43 @@ public function delete($source) {
    * {@inheritdoc}
    */
   public function setup() {
-    // Get index definition.
-    $index_definition = $this->getIndexDefinition();
+    $operation = ElasticsearchOperations::INDEX_CREATE;
 
     // Create one index per language, so that we can have different analyzers.
     foreach ($this->language_manager->getLanguages() as $langcode => $language) {
-      // Get index name.
-      $index_name = $this->getIndexName(['langcode' => $langcode]);
+      // Get index definition.
+      $index_definition = $this->getIndexDefinition(['langcode' => $langcode]);
+
+      try {
+        // Get index name.
+        $index_name = $this->getIndexName(['langcode' => $langcode]);
 
-      // Get analyzer for the language.
-      $analyzer = ElasticsearchLanguageAnalyzer::get($langcode);
+        // Get analyzer for the language.
+        $analyzer = ElasticsearchLanguageAnalyzer::get($langcode);
 
-      // Put analyzer parameter to all "text" fields in the mapping.
-      foreach ($index_definition->getMappingDefinition()->getProperties() as $property) {
-        if ($property->getDataType()->getType() == 'text') {
-          $property->addOption('analyzer', $analyzer);
+        // Put analyzer parameter to all "text" fields in the mapping.
+        foreach ($index_definition->getMappingDefinition()->getProperties() as $property) {
+          if ($property->getDataType()->getType() == 'text') {
+            $property->addOption('analyzer', $analyzer);
+          }
         }
-      }
 
-      if (!$this->client->indices()->exists(['index' => $index_name])) {
-        $this->client->indices()->create([
-          'index' => $index_name,
-          'body' => $index_definition->toArray(),
-        ]);
+        if (!$this->client->indices()->exists(['index' => $index_name])) {
+          $request_params = [
+            'index' => $index_name,
+            'body' => $index_definition->toArray(),
+          ];
+
+          // Create the index.
+          $callback = [$this->client->indices(), 'create'];
+          $request_event = $this->dispatchOperationRequestEvent($operation, $callback, $request_params);
+
+          $result = call_user_func_array($request_event->getCallback(), $request_event->getCallbackParameters());
+          $this->dispatchOperationResultEvent($result, $operation, NULL, $request_params);
+        }
+      } catch (\Exception $e) {
```

Comment:
We could perhaps catch `Throwable` here instead so it would catch both errors and exceptions and not just exception.